java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,602 @@
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL_TO:
+      case EQUAL_TO:
+      case NOT_EQUAL_TO:
+        BinaryExpressionTree binaryExpressionTree = (BinaryExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(binaryExpressionTree.rightOperand());
+        build(binaryExpressionTree.leftOperand());
+        break;
+      case ASSIGNMENT:
+        AssignmentExpressionTree assignmentExpressionTree = (AssignmentExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(assignmentExpressionTree.variable());
+        build(assignmentExpressionTree.expression());
+        break;
+      case MEMBER_SELECT:
+        MemberSelectExpressionTree mse = (MemberSelectExpressionTree) tree;
+        currentBlock.elements.add(mse);
+        build(mse.expression());
+        break;
+      case CONDITIONAL_AND: {
+        BinaryExpressionTree e = (BinaryExpressionTree) tree;
+        // process RHS
+        Block falseBlock = currentBlock;
+        currentBlock = createBlock(falseBlock);
+        build(e.rightOperand());
+        Block trueBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        build(e.leftOperand());
+        break;
+      }
+      case CONDITIONAL_OR: {
+        BinaryExpressionTree e = (BinaryExpressionTree) tree;
+        // process RHS
+        Block trueBlock = currentBlock;
+        currentBlock = createBlock(trueBlock);
+        build(e.rightOperand());
+        Block falseBlock = currentBlock;
+        // process LHS
+        currentBlock = createBranch(e, trueBlock, falseBlock);
+        build(e.leftOperand());
+        break;
+      }
+      case LABELED_STATEMENT: {
+        LabeledStatementTree s = (LabeledStatementTree) tree;
+        build(s.statement());
+        currentBlock = createBlock(currentBlock);
+        labels.put(s.label().name(), currentBlock);
+        break;
+      }
+      case SWITCH_STATEMENT: {
+        // FIXME useless node created for default cases.
+        SwitchStatementTree switchStatementTree = (SwitchStatementTree) tree;
+        Block switchSuccessor = currentBlock;
+        // process condition
+        currentBlock = createBlock();
+        currentBlock.terminator = switchStatementTree;
+        switches.addLast(currentBlock);
+        build(switchStatementTree.expression());
+        // process body
+        currentBlock = createBlock(switchSuccessor);
+        breakTargets.addLast(switchSuccessor);
+        if (!switchStatementTree.cases().isEmpty()) {
+          CaseGroupTree firstCase = switchStatementTree.cases().get(0);
+          for (CaseGroupTree caseGroupTree : switchStatementTree.cases()) {
+            build(caseGroupTree.body());
+            switches.getLast().successors.add(currentBlock);
+            if (caseGroupTree != firstCase) {
+              // No block predecessing the first case group.
+              currentBlock = createBlock(currentBlock);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 17. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)